### PR TITLE
get all root blocks, not just the first one

### PR DESCRIPTION
**Pull request by koubaa**
LoadAEDTFile was ignored all but the first begin/end block.  We need the last one in order to extract the preview image

### CODE REVIEW

File: pyaedt/generic/LoadAEDTFile.py [hunk 1]
```diff
@@ -161,17 +161,15 @@ def _decode_value_and_save(k, v, d):
             d[k] = _parse_value(v)
 
 
-def _decode_key(l, d):
+def _decode_key(l: str, d) -> None:
```
Comment (akaszynski) [inline pyaedt/generic/LoadAEDTFile.py:164]: Type checking isn't permitted since we need to import on IronPython 2.7

File: pyaedt/generic/LoadAEDTFile.py [hunk 2]
```diff
@@ -161,17 +161,15 @@ def _decode_value_and_save(k, v, d):
...
     """
 
     Parameters
     ----------
-    l :
-
-    d :
+    l : line
```
Comment (akaszynski) [inline pyaedt/generic/LoadAEDTFile.py:169]:
```suggestion
    l : str
        Line.
```

File: pyaedt/generic/LoadAEDTFile.py [hunk 3]
```diff
@@ -161,17 +161,15 @@ def _decode_value_and_save(k, v, d):
...
 
+    d : active_dictionary
```
Comment (akaszynski) [inline pyaedt/generic/LoadAEDTFile.py:171]:
```suggestion
    d : str
        Active dictionary.
```

File: pyaedt/generic/LoadAEDTFile.py [hunk 4]
```diff
@@ -304,17 +307,16 @@ def _load_entire_aedt_file(filename):
         dictionary containing the decoded AEDT file
 
     """
-    _reaf_aedt_file(filename)
+    global _count
```
Comment (akaszynski) [inline pyaedt/generic/LoadAEDTFile.py:310]: There must be a way to avoid use of globals.

File: pyaedt/generic/LoadAEDTFile.py [hunk 5]
```diff
@@ -304,17 +307,16 @@ def _load_entire_aedt_file(filename):
...
+        line = _all_lines[_count]
         m = _begin_search.search(line)
         if m:
-            begin_key = m.group(1)
-            break
-    main_dict = OrderedDict()
-    if begin_key:
-        _walk_through_structure(begin_key, main_dict)
+            _walk_through_structure(m.group(1), main_dict)
+        _count = _count + 1
```
Comment (akaszynski) [inline pyaedt/generic/LoadAEDTFile.py:319]: This is a bit cleaner.  Performance gains are marginal on Python 3.8 (local), not sure on IronPython

```suggestion
        _count += 1
```

File: pyaedt/generic/LoadAEDTFile.py (hunk 4 above)
Comment (koubaa) [reply pyaedt/generic/LoadAEDTFile.py:310]: @akaszynski I can easily do it but if I'm going to make that sort of change I'll want to unit test it first.  @maxcapodi78 two questions, 1) how to run cpython unit tests 2) is it ok if the tests for this file are cpython only (I don't think it needs to run in both contexts)

Comment (akaszynski) [reply pyaedt/generic/LoadAEDTFile.py:310]: Unit test documentation is here:
https://aedtdocs.pyansys.com/Resources/Contributing.html#testing

Our coverage should be testing this.  Check coverage locally by appending `--cov pyaedt --cov-report annotate` to the `pytest` command.

Comment (akaszynski) [reply pyaedt/generic/LoadAEDTFile.py:310]: global refactor can wait until a different PR since this is out of scope of this PR.

Comment (koubaa) [reply pyaedt/generic/LoadAEDTFile.py:310]: @akaszynski 
ERROR: usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: unrecognized arguments: --cov --cov-report annotate _unittest
  inifile: C:\AnsysDev\NoBackup\code\ThirdPartySource\PyAEDT\pytest.ini
  rootdir: C:\AnsysDev\NoBackup\code\ThirdPartySource\PyAEDT

File: pyaedt/generic/LoadAEDTFile.py (hunk 1 above)
Comment (maxcapodi78) [reply pyaedt/generic/LoadAEDTFile.py:164]: Yes, please revert it. 

```suggestion
def _decode_key(l, d):
```

File: pyaedt/generic/LoadAEDTFile.py (hunk 4 above)
Comment (maxcapodi78) [reply pyaedt/generic/LoadAEDTFile.py:310]: @koubaa all Unit Tests are in 2 folders _unittest and _unittest_ironpython. If you run on PyCharm simply configure and install pytest and run it from there (right click on folder/file). for UnitTest is slightly different because they run as batch command within AEDT. They both runs on CICD

**Review by maxcapodi78 — CHANGES_REQUESTED** [review]

**Comment by koubaa** [comment]
@maxcapodi78 I added specific unit tests for load_entire_aedt_file that tests a few scenarios

File: _unittest/test_35_LoadAEDTFile.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+# Import required modules
+from pyaedt.generic.filesystem import Scratch
+from pyaedt.generic.LoadAEDTFile import load_entire_aedt_file
+import base64
+import filecmp
+import os
+from typing import Dict
+
+
+def _write_jpg(design_info: Dict, scratch: str) -> str:
```
Comment (akaszynski) [inline _unittest/test_35_LoadAEDTFile.py:13]:
```suggestion
def _write_jpg(design_info, scratch):
```

File: _unittest/test_35_LoadAEDTFile.py [hunk 7]
```diff
@@ -0,0 +1,73 @@
+# Setup paths for module imports
+from _unittest.conftest import local_path, scratch_path
+
+# Import required modules
+from pyaedt.generic.filesystem import Scratch
+from pyaedt.generic.LoadAEDTFile import load_entire_aedt_file
+import base64
+import filecmp
+import os
+from typing import Dict
```
Comment (akaszynski) [inline _unittest/test_35_LoadAEDTFile.py:10]:
```suggestion
```

**Comment by akaszynski** [comment]
Nevermind, deleted two comments; we've separated out IronPython unit tests from CPython unit tests.

**Review by akaszynski — APPROVED** [review]
LGTM.  Thanks for adding in the "semi-edge case" unit tests.

**Comment by koubaa** [comment]
> LGTM. Thanks for adding in the "semi-edge case" unit tests.

@akaszynski I had to follow the boy scout rule, leaving it in better shape than I found it.
